File: C/Two Pointer/392. Is Subsequence.c
/* 392. Is Subsequence
Difficulty: Medium
Submission Link: https://leetcode.com/problems/is-subsequence/submissions/1246684124/ */

bool isSubsequence(char* s, char* t) {
    while(*s && *t) {
        if(*s == *t) {
            s++;
        }
        t++;
    }
    return *s == '\0';
}

File: JavaScript/Two Pointer/392. Is Subsequence.js
/* 392. Is Subsequence
Difficulty: Medium
Submission Link: https://leetcode.com/problems/is-subsequence/submissions/1246682827/
*/

var isSubsequence = function(s, t) {
    let i = 0, j = 0;
    while (i < s.length && j < t.length) {
        if (s[i] === t[j]) {
            i++;
        }
        j++;
    }
    return i === s.length;
}